Google.Api.Generator/Generation/SnippetCodeGenerator.cs

```diff
@@ -699,38 +728,45 @@ public static IEnumerable<ResourceName> Create(Signature sig)
                         (object)_sig._def.Client.Call(Method.SyncMethodName)(InitRequestArgs.ToArray()) :
                         _sig._def.Response.WithInitializer(_sig._def.Client.Call(Method.SyncMethodName)(InitRequestArgs.ToArray())));
 
-                    public SnippetDef SyncSnippet => new SnippetDef(Method, true, SyncMethodName, ctx => WithSourceFileContext(ctx).SyncMethod);
+                    public SnippetDef SyncSnippet => new SnippetDef(Method, true, SyncMethodName,
```

Comment:
Likewise I suspect another abstraction here could simplify things... but I suspect I'd have to create a separate PR to try it.